plugins/filter/list.py

```diff
@@ -1,43 +1,116 @@
 # -*- coding: utf-8 -*-
-# Copyright (c) 2020, Vladimir Botka <vbotka@gmail.com>
+# Copyright (c) 2020-2022, Vladimir Botka <vbotka@gmail.com>
 # GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
 
 from __future__ import (absolute_import, division, print_function)
 __metaclass__ = type
 
-from ansible.errors import AnsibleError, AnsibleFilterError
+from ansible.errors import AnsibleFilterError
 from ansible.module_utils.six import string_types
 from ansible.module_utils.common._collections_compat import Mapping, Sequence
+from ansible.utils.vars import merge_hash
+from ansible.release import __version__ as ansible_version
+
 from collections import defaultdict
 from operator import itemgetter
+from packaging.specifiers import SpecifierSet
+from packaging.version import Version
```

Comment:
Please use `LooseVersion` from `ansible_collections.community.general.plugins.module_utils.version` instead. `packaging` is not necessarily installed everywhere (it has been a requirement since 2.10, but is not one of 2.9).